### ISSUE TITLE
nx.to_pandas_edgelist() will not update the "source" and "target" columns if they already exist

### ISSUE THREAD

**Issue by iboates** [networkx/networkx#3930]
I discovered this during a workflow where I am transforming between pandas dataframes and networkx graphs rather frequently. The old "source" and "target" columns were hanging around from a previous transformation.

I realize that it is probably bad practice to already have "source" and "target" as attributes in a graph, especially when they are wrong, but there is no mention of this behaviour on the [docs page](https://networkx.github.io/documentation/stable/reference/generated/networkx.convert_matrix.to_pandas_edgelist.html) and it seems to me that a more intuitive behaviour would be for it to overwrite the columns in the dataframe with the actual current graph information, and not what may be hanging around in the attributes list.

See below. In this example, I create an undirected graph with a "backwards" edge, then straighten the graph out by transforming it into a tree [(see here)](https://stackoverflow.com/questions/60742779/creating-a-directed-graph-from-an-undirected-graph-with-dfs-tree-but-keep-attri). However, since "source" and "target" already exist (perhaps from a previous conversion from a graph to dataframe, and then back to a graph again), then they are not updated when writing back to a dataframe again.

```
import networkx as nx

graph = nx.Graph([
    (0, 1, {"source": 0, "target": 1}),
    (1, 2, {"source": 1, "target": 2}),
    (3, 2, {"source": 3, "target": 2})
])
digraph = nx.DiGraph(graph)
digraph.remove_edges_from(digraph.edges - nx.dfs_edges(graph, 0))

df1 = nx.to_pandas_edgelist(digraph)  # Default is to use source="source" and target="target"
print(df1.head())
df2 = nx.to_pandas_edgelist(digraph, source="source2", target="target2")
print(df2.head())
```

Output of ´print(df1.head())´ is:

```
   source  target
0       0       1
1       1       2
2       3       2
```

Output of ´print(df2.head())´ is:

```
   source2  target2  source  target
0        0        1       0       1
1        1        2       1       2
2        2        3       3       2
```

**Comment by dschult** [comment]
Yup -- it looks like the code actually does write "source" and "target" to the edge attributes and then overwrites them with the previously existing data.  Maybe it should raise an exception if the columns already exist. It's not good to overwrite data without telling someone.

**Comment by iboates** [comment]
[This](https://github.com/networkx/networkx/blob/7ae7f0146a33e340acbabe5a11d297e8187cb6d8/networkx/convert_matrix.py#L214) is the relevant code, yes? I can implement an exception raise. Should it just be something generic, like a `ValueError` or should it be one of the specialized exceptions defined in `exception.py`?

**Comment by dschult** [comment]
Yes -- that's the function... line 262 is where it gets overwritten, but the keys could be checked a few lines before... 

I think the general ```NetworkXError``` makes sense here -- 
that flags it as an exception of the NX api and I don't think it 
makes sense to create a special exception  just for this case.

Thanks!

**Comment by iboates** [comment]
I made a PR here: https://github.com/networkx/networkx/pull/3935

**Comment by dschult** [comment]
Fixed in #3935
